pallets/vault-registry/src/types.rs

```diff
@@ -660,27 +659,17 @@ impl<T: Config> RichVault<T> {
 
 	fn new_deposit_public_key(
 		&self,
-		secure_id: H256,
+		_secure_id: H256,
 	) -> Result<StellarPublicKeyRaw, DispatchError> {
-		// TODO fix me
-		// let vault_public_key = Pallet::<T>::get_bitcoin_public_key(&self.data.id.account_id)?;
-		// let vault_public_key = vault_public_key
-		// 	.new_deposit_public_key(secure_id)
-		// 	.map_err(|_| Error::<T>::InvalidPublicKey)?;
-
-		let vault_public_key: StellarPublicKeyRaw = [0u8; 32];
-		Ok(vault_public_key)
+		// The new deposit public key will always be the same Vault Public key.
+		Pallet::<T>::get_stellar_public_key(&self.data.id.account_id)
 	}
 
 	pub(crate) fn new_deposit_address(
 		&mut self,
 		secure_id: H256,
 	) -> Result<StellarPublicKeyRaw, DispatchError> {
-		let public_key = self.new_deposit_public_key(secure_id)?;
-		// let btc_address = BtcAddress::P2WPKHv0(public_key.to_hash());
-		let stellar_address: StellarPublicKeyRaw = [0; 32];
-		// TODO change this to a stellar address
-		Ok(stellar_address)
+		self.new_deposit_public_key(secure_id)
 	}
```

Comment:
Maybe we should completely remove these functions since we don't use deposit addresses and it might be confusing. Although keeping it like this is also okay since `new_deposit_public_key` has the comment that makes it clear. WDYT @gonzamontiel ?

Reply:
I thought about removing them, my thought after was to keep that abstraction layer and add the comment. It can be a bit confusing or a newcomer, and If we never ever plan to adapt the deposit address in any way, then that function does not make sense and I would remove it.

Reply:
Yes, I agree. But I doubt we will ever use/need something like this so I would vote for removing it.